client/client-multi/client-c/src/platform-impl/kaa_tcp_channel.c

```diff
@@ -869,11 +916,14 @@ void kaa_tcp_channel_connack_message_callback(void *context, kaatcp_connack_t me
 void kaa_tcp_channel_disconnect_message_callback(void *context, kaatcp_disconnect_t message)
 {
     KAA_RETURN_IF_NIL(context,);
-    kaa_tcp_channel_t *channel = (kaa_tcp_channel_t *) context;
 
-    KAA_LOG_INFO(channel->logger, KAA_ERR_NONE,"Kaa TCP channel [0x%08X] DISCONNECT message received"
-                                                                            , channel->access_point.id);
-    kaa_tcp_channel_socket_io_error(channel);
+    KAA_LOG_INFO(((kaa_tcp_channel_t *) context)->logger, KAA_ERR_NONE,"Kaa TCP channel [0x%08X] DISCONNECT message received"
+                                                                            , ((kaa_tcp_channel_t *) context)->access_point.id);
+
+    if (message.reason != KAATCP_DISCONNECT_INTERNAL_ERROR) {
+        ((kaa_tcp_channel_t *) context)->sync_state = KAA_TCP_CHANNEL_SYNC_OP_STARTED;
+    }
+    kaa_tcp_channel_socket_io_error(((kaa_tcp_channel_t *) context));
```

Comment:
Do we really need to call "kaa_tcp_channel_socket_io_error" each time on the DISCONNECT callback?